### PR TITLE
Adding link detector regex to support c# extension

### PR DESCRIPTION
**Pull request by rajkumar42**
Adding link detector regex for file links of the following format

at app1.Program.Foo() in c:\temp\dotnet\app1\Program.cs:line 9

### CODE REVIEW

**Comment by rajkumar42** [comment]
Please review @gregg-miskelly @isidorn

**Comment by gregg-miskelly** [comment]
Since our matching at the end of the string isn't very strong, I think we might want to tighten up the start of path matching to only support Drive-colon (`[a-zA-Z]:\`) or Windows network share syntax (`\\`) and Unix path syntax (`/`). Ex: `Foo/Program.cs:line 9` shouldn't match

**Comment by gregg-miskelly** [comment]
Using `\w` to match the characters in `line` isn't ideal as -- `\w` includes numbers, which we definitely don't want, and it sounds like it doesn't include accented characters. Can you try `\p{L}`?

**Comment by gregg-miskelly** [comment]
Two other thoughts:
1. Seems like the line number should be required (no `?`) at the end
2. Seems like we should require it to be the line ending (`$` at the end)

**Comment by michelkaporin** [comment]
Agree with @gregg-miskelly comments.

Is it possible to also address #28508 for c# regex while you are on it? This regex suffers of the same problem. 

Besides that it looks good to me.

File: src/vs/workbench/parts/debug/browser/linkDetector.ts [hunk 1]
```diff
@@ -19,9 +19,30 @@ export class LinkDetector {
...
+        // Regular Expression for filepaths from C# extension
+		// at app1.Program.Foo() in c:\temp\dotnet\app1\Program.foo.cs:line 9
+        // at app1.Program.Foo() in c:\temp folder\dotnet\app1\Program.cs:line 9
+        // at app3.Program.Main(String[] args) in /Users/janraj/temp/app3/Program.cs:line 9
+		new RegExp("(" +
+						"(?:([a-zA-Z]:)|[\\]*|[\/]+)" + // Rooted file starting with c: or \\ or /
+						"(?:[\\/][^<>\\/\'\"\[\]:]*)+"+ // File path
+					")" +
+					":line\ " +	// ends with :line literal
+					"(\d+)", // Line number
```
Comment (gregg-miskelly) [inline src/vs/workbench/parts/debug/browser/linkDetector.ts]: I might suggest adding a '$' also for end-of-line

**Review by gregg-miskelly — APPROVED** [review]
Assuming that the first RegEx only needs to match full paths, LGTM.

**Comment by michelkaporin** [comment]
My VS Code instance freezes with the latest changes.

Here is the code I am testing it with, if it helps:

```
console.log('http://test.com:8080');
console.log('Relative paths:');
console.log('.vscode\\launch.json:5:11'); // relative path
console.log('.vscode/launch.json:5:11'); // relative path
console.log('./linkDetection.ts:5:11');
console.log('linkDetection.ts:5:11');
console.log('linkDetection.ts:15');
console.log('test (src/command/AsyncCommand.ts:6:24)');
console.log('Full paths:');
console.log('c:/Users/t-mikapo/Documents/Projects/multiline-object-properties/out/linkDetection.js:5:11');
console.log('c:\\Users\\t-mikapo\\Documents\\Projects\\multiline-object-properties\\out\\linkDetection.js:5:11');
console.log('c:/Users/t-mikapo/Documents/Projects/multiline-object-properties/out/linkDetection.js:5');
console.log('c:\\Users\\t-mikapo\\Documents\\Projects\\multiline-object-properties\\out\\linkDetection.js:5');
console.log('Other:');
console.log('c:/Users/t-mikapo/Documents/Projects/multiline-object-properties/out/linkDetection.js:5:11 combined with ./linkDetection.ts:5)');
console.log('at model.newPostInstance (.vscode/launch.json:11:3)');
console.log('at model.newPostInstance (.vscode/launch.json on line 8, column 13)');
console.log('at model.newPostInstance (.vscode/launch.json:(45,18))');
console.log('d:\\proj\\clicon\\Program.cs:line 23');
console.log('d:\\proj\\clicon\\Program.cs:line 23, column 13');
console.log('d:\\proj\\clicon\\Program.cs line 23, column 13');
console.log('file:\\\c:\\Users\\t-mikapo\\Documents\\Projects\\multiline-object-properties\\out\\linkDetection.js:5:11');
```

**Comment by michelkaporin** [comment]
@rajkumar42 with your latest commit it does not match C# syntax any more. Please verify your changes before commiting, and let me know once ready for a final review 👍

**Comment by rajkumar42** [comment]
@michelkaporin I couldn't reproduce the issue, my testing is limited to mac. I've pushed a change from single line to multiline. Can you please check if that fixes the issue?

**Comment by michelkaporin** [comment]
@isidorn would you be able to do a final review of that as I am out please? I was using the code above to test this PR on Windows.

**Comment by gregg-miskelly** [comment]
Ping

**Comment by isidorn** [comment]
Thanks for your PR and sorry that I am only now looking into this.
This PR is very C# specific, a better approach to solving this would be to allow extensions to somehow contribute different link detection mechanisms.
However currently all our link detection is not in the best state and rather than accepting PRs that complicate the situation further I would prefer that we somehow restrucutre and improve our Link detection. For that I have created a debt item https://github.com/Microsoft/vscode/issues/34026

Due to the reasons above I will not accept this PR.

A PR which I would be willing to accept is one that is language agnostic and has a less cryptic structure - not a complex regex which seems very hard to maintain and modify imho

**Pull request closed without merging**